docs/manifests_and_customizing_via_api.md

```diff
@@ -0,0 +1,335 @@
+  subnets:
+  - us-east-2d
+  - us-east-2c
+```
+
+This configuration will create an autoscale group that will include 42 m4.10xlarge nodes running as spot instances with custom labels.
+
+To create the cluster execute:
+
+```
+kops create -f $NAME.yaml
+kops create secret --name $NAME sshpublickey admin -i ~/.ssh/id_rsa.pub
+kops update cluster $NAME --yes
+```
+
+Update the cluster spec YAML file, and to update the cluster run:
+
+```
+kops replace -f $NAME.yaml
+kops update cluster $NAME --yes
+```
+
+## Further References
+
+`kops` implements a full API that defines the various elements in the YAML file exported above.  Two top level components exist; `ClusterSpec` and `InstanceGroup`.
+
+### Cluster Spec
+
+```YAML
+apiVersion: kops/v1alpha2
+kind: Cluster
+metadata:
+  creationTimestamp: 2017-05-04T23:21:47Z
+  name: k8s.example.com
+spec:
+  api:
+```
+
+Full documentation is accessible via [godoc](https://godoc.org/k8s.io/kops/pkg/apis/kops#ClusterSpec).
+
+The `ClusterSpec` allows a user to set configurations for such values as Docker log driver, Kubernetes API server log level, VPC for reusing a VPC (`NetworkID`), and the Kubernetes version.
+
+More information about some of the elements in the `ClusterSpec` is available in the following:
+
+-  Cluster Spec [document](cluster_spec.md) which outlines some of the values in the Cluster Specification.
+- [Ectd Encryption](etcd_backup.md)
+- [GPU](gpu.md) setup
+- [IAM Roles](iam_roles.md) - adding additional IAM roles.
+- [Labels](lables.md)
+- [Run In Existing VPC](run_in_existing_vpc.md)
+
+To access the full configuration that a `kops` installation is running execute:
+
+```
+kops get cluster $NAME --full -o yaml
+```
+
+This command prints the entire YAML configuration.  But _do not_ use the full document, you may experience strange and unique unwanted behaviors. 
+
+### Instance Groups
+
+```YAML
+apiVersion: kops/v1alpha2
+kind: InstanceGroup
+metadata:
+  creationTimestamp: 2017-05-04T23:21:48Z
+  name: foo
+spec:
+```
+
+Full documentation is accessible via [godocs] (https://godoc.org/k8s.io/kops/pkg/apis/kops#InstanceGroupSpec).
+
+Instance Groups map to auto scale groups in AWS, and Instance Groups in GCE.  They are an API level description of a group of compute instances used as Masters or Nodes.
+
+More documentation is available in the [Instance Group](instance_groups.md) document.
+
+## Closing Thoughts
+
+Using a document based configuration for building and managing `kops` clusters is powerful.  Do use these features with caution.  If you do not need to define or customize a value, let `kops` create and possibly modify the value by default.  Again do not use the full ClusterSpec.  Users have wasted days debugging issues, due to using the full specification.  But if you need to run a custom version of Kubernetes Controller Manager, set `kubeControllerManager.image` and update your cluster.  Do take care with setting custom values and test before using in production.
```

Comment:
This might be confusing `Users have wasted days debugging issues, due to using the full specification.`, maybe give an example showing the `--full` arg and explain what it does and why it's a bad idea?